Hosting Logs: remove `atomic_site_id`

oandregal opened the pull request:
Follow-up to https://github.com/Automattic/wp-calypso/pull/99056

## Proposed Changes

Removes the "Atomic Site ID" from the Fields API and the download logs.

## Why are these changes being made?

This is an implementation detail that is not useful to users.

## Testing Instructions

- Go to PHP Logs, and open the details modal. Verify the "Atomic Site ID" is not present there.
- Download the PHP Logs and verify the "atomic_site_id" field is not present and all the rest work fine.

oandregal committed:
Remove from fields

oandregal committed:
Remove atomic site id from download

tyxla reviewed (commented):
This works well when downloading the data. However, in the log details modal, I'm still seeing an "Atomic Site ID":

![Screenshot 2025-03-05 at 14 46 13](https://github.com/user-attachments/assets/d7cf5c67-f522-49a0-b72e-3ae4a8cadaf3)

Haven't tested, but I think we'll need to remove this from the modal:

```diff
diff --git a/client/sites/logs/components/details-modal-php.tsx b/client/sites/logs/components/details-modal-php.tsx
index 6752d77dcb7..0335e52586f 100644
--- a/client/sites/logs/components/details-modal-php.tsx
+++ b/client/sites/logs/components/details-modal-php.tsx
@@ -50,8 +50,6 @@ const DetailsModalPHP = ( { item }: DetailsModalPHPProps ) => {
                        <div>{ item.file }</div>
                        <div className="site-logs-details-modal__field-title">{ translate( 'LINE' ) }</div>
                        <div>{ numberFormat( item.line ) }</div>
-                       <div className="site-logs-details-modal__field-title">{ translate( 'ATOMIC SITE ID' ) }</div>
-                       <div>{ item.atomic_site_id }</div>
                </div>
        );
 };
```

Furthermore, have you considered removing the field at the query level? Like this:

```diff
diff --git a/client/data/hosting/use-site-logs-query.ts b/client/data/hosting/use-site-logs-query.ts
index 4d2ad8faf9c..c347e11f184 100644
--- a/client/data/hosting/use-site-logs-query.ts
+++ b/client/data/hosting/use-site-logs-query.ts
@@ -142,7 +142,10 @@ export function useSiteLogsQuery(
                                has_more: !! data.scroll_id,
                                total_results:
                                        typeof data.total_results === 'number' ? data.total_results : data.total_results.value,
-                               logs: data.logs.map( ( log, key ) => ( { ...log, id: String( key ) } ) ),
+                               logs: data.logs.map( ( { atomic_site_id, ...log }, key ) => ( {
+                                       ...log,
+                                       id: String( key ),
+                               } ) ),
                                scroll_id: data.scroll_id,
                        };
                },
```

oandregal committed:
Remove atomic site ID from modal

oandregal committed:
Remove atomic_site_id at the query level

oandregal commented:
> This works well when downloading the data. However, in the log details modal, I'm still seeing an "Atomic Site ID":

Fixed this. 

> Furthermore, have you considered removing the field at the query level? Like this:

I like this thinking :)

In practice, it doesn't solve anything because 1) the downloads do their own separate thing and 2) the field still has to be removed so it's not registered. However, I like the idea that this is baked in intentionally, so our future-selves don't add it back unintentionally — so I've gone ahead and implemented it.

tyxla reviewed (approved):
This tests well and the code looks good. Thanks @oandregal 🚀

oandregal committed:
Prettify files after rule change
